config/initializers/devise.rb

```diff
@@ -311,3 +311,21 @@
   # config.omniauth_path_prefix = '/my_engine/users/auth'
 
 end
+
+Devise.saml_update_resource_hook = Proc.new do |user, saml_response, auth_value|
+  # Copied from the default behavior: https://github.com/apokalipto/devise_saml_authenticatable/blob/09221d59e434bc947327e478a388013a70add4f7/lib/devise_saml_authenticatable.rb#L101
+  saml_response.attributes.resource_keys.each do |key|
+    user.send "#{key}=", saml_response.attribute_value_by_resource_key(key)
+  end
+
+  if Devise.saml_use_subject
+    user.send "#{Devise.saml_default_user_key}=", auth_value
+  end
+
+  puts "User: #{user.inspect}"
```

Comment:
**Question** Is this statement intentional?

Reply:
Sorry @smritigarg, I added this for debugging and forgot to remove it before committing the changes.